ADO.NET/OAuthSampleApp/OAuthSampleApp.cs

```diff
@@ -0,0 +1,217 @@
+﻿using Vertica.Data.VerticaClient;
+using Newtonsoft.Json.Linq;
+using System.Configuration;
+using System.Runtime.InteropServices;
+using System.ComponentModel;
+
+internal class OAuthSampleApp
+{
+    private static VerticaConnectionStringBuilder connectionStringBuilder;
+    private static VerticaConnection vConnection;
+    private const string OAUTH_ACCESS_TOKEN_VAR_STRING = "OAUTH_SAMPLE_ACCESS_TOKEN";
+    private const string OAUTH_REFRESH_TOKEN_VAR_STRING = "OAUTH_SAMPLE_REFRESH_TOKEN";
+
+    static async Task Main(string[] args)
+    {
+        connectionStringBuilder = new VerticaConnectionStringBuilder();
+        connectionStringBuilder.ConnectionString = ConfigurationManager.ConnectionStrings["ConnectionString"].ConnectionString;
+
+        await SetUp();
+        await EnsureAccessToken();
+        await ConnectToDatabase();
+        await TearDown();
+    }
+
+    // admin must log in and setup the user for OAuth
+    // steps can also be followed in vsql
+    static async Task SetUp()
+    {
+        vConnection = new VerticaConnection(connectionStringBuilder.ConnectionString);
+        vConnection.Open();
+
+        string USER = ConfigurationManager.AppSettings["User"];

```

Comment:
It looks like we pull from AppSettings a lot. Consider replacing with

```
static getSetting(String setting)
{
    return ConfigurationManager.AppSettings[setting]
}
```

Reply:
Done.